check.go

```diff
@@ -149,4 +190,26 @@ func init() {
 		charge.CategoryName = c.Attr("", "description")
 		return nil
 	})
+
+	// Scan price-1.2 extension into Charges
+	path = "epp > response > extension > " + ExtPrice + " chkData"
+	scanResponse.MustHandleStartElement(path+">cd", func(c *xx.Context) error {
+		dcd := &c.Value.(*response_).DomainCheckResponse
+		dcd.Charges = append(dcd.Charges, DomainCharge{})
+		return nil
+	})
+	scanResponse.MustHandleCharData(path+">cd>name", func(c *xx.Context) error {
+		charges := c.Value.(*response_).DomainCheckResponse.Charges
+		charge := &charges[len(charges)-1]
+		charge.Domain = string(c.CharData)
+		return nil
+	})
+	scanResponse.MustHandleCharData(path+">cd>category", func(c *xx.Context) error {
+		charges := c.Value.(*response_).DomainCheckResponse.Charges
+		charge := &charges[len(charges)-1]
+		if strings.Index(strings.ToLower(string(c.CharData)), "premium") != -1 {
```

Comment:
It seems like the categories aren't very standardized. For instance, a regular domain is `STANDARD`, while the few premiums I saw used `Premium_` with numbers added.

I realize the code is more complicated, I figured it was worth making it more robust.